cmplxs/cmplxs.go

```diff
@@ -0,0 +1,679 @@
+	c128.AddConst(c, dst)
+}
+
+// AddScaled performs dst = dst + alpha * s.
+// It panics if the lengths of dst and s are not equal.
+func AddScaled(dst []complex128, alpha complex128, s []complex128) {
+	if len(dst) != len(s) {
+		panic("cmplxs: length of destination and source to not match")
+	}
+	c128.AxpyUnitaryTo(dst, alpha, s, dst)
+}
+
+// AddScaledTo performs dst = y + alpha * s, where alpha is a scalar,
+// and dst, y and s are all slices.
+// It panics if the lengths of dst, y, and s are not equal.
+//
+// At the return of the function, dst[i] = y[i] + alpha * s[i]
+func AddScaledTo(dst, y []complex128, alpha complex128, s []complex128) []complex128 {
+	if len(dst) != len(s) || len(dst) != len(y) {
+		panic("cmplxs: lengths of slices do not match")
+	}
+	c128.AxpyUnitaryTo(dst, alpha, s, y)
+	return dst
+}
+
+// Count applies the function f to every element of s and returns the number
+// of times the function returned true.
+func Count(f func(complex128) bool, s []complex128) int {
+	var n int
+	for _, val := range s {
+		if f(val) {
+			n++
+		}
+	}
+	return n
+}
+
+// CumProd finds the cumulative product of the first i elements in
+// s and puts them in place into the ith element of the
+// destination dst. A panic will occur if the lengths of arguments
+// do not match.
+//
+// At the return of the function, dst[i] = s[i] * s[i-1] * s[i-2] * ...
+func CumProd(dst, s []complex128) []complex128 {
+	if len(dst) != len(s) {
+		panic("cmplxs: length of destination does not match length of the source")
+	}
+	if len(dst) == 0 {
+		return dst
+	}
+	return c128.CumProd(dst, s)
+}
+
+// CumSum finds the cumulative sum of the first i elements in
+// s and puts them in place into the ith element of the
+// destination dst. A panic will occur if the lengths of arguments
+// do not match.
+//
+// At the return of the function, dst[i] = s[i] + s[i-1] + s[i-2] + ...
+func CumSum(dst, s []complex128) []complex128 {
+	if len(dst) != len(s) {
+		panic("cmplxs: length of destination does not match length of the source")
+	}
+	if len(dst) == 0 {
+		return dst
+	}
+	return c128.CumSum(dst, s)
+}
+
+// Distance computes the L-norm of s - t. See Norm for special cases.
+// A panic will occur if the lengths of s and t do not match.
+func Distance(s, t []complex128, L float64) float64 {
+	if len(s) != len(t) {
+		panic("cmplxs: slice lengths do not match")
+	}
+	if len(s) == 0 {
+		return 0
+	}
+	if L == 2 {
+		return c128.L2DistanceUnitary(s, t)
+	}
+	var norm float64
+	if L == 1 {
+		for i, v := range s {
+			norm += cmplx.Abs(t[i] - v)
+		}
+		return norm
+	}
+	if math.IsInf(L, 1) {
+		for i, v := range s {
+			absDiff := cmplx.Abs(t[i] - v)
+			if absDiff > norm {
+				norm = absDiff
+			}
+		}
+		return norm
+	}
+	for i, v := range s {
+		norm += math.Pow(cmplx.Abs(t[i]-v), L)
+	}
+	return math.Pow(norm, 1/L)
+}
+
+// Div performs element-wise division dst / s
+// and stores the value in dst. It panics if the
```

Comment:
"values" or "result"?

Reply:
Done

Will also need to be done in floats.